rust/lance/src/index/vector.rs

```diff
@@ -65,6 +65,10 @@ pub struct VectorIndexParams {
 
     /// Vector distance metrics type.
     pub metric_type: MetricType,
+
+    /// Use V3 Index builder.
+    /// Only used by IVF_PQ index since IVF_PQ still creates old index format by default.
+    pub force_use_new_index_format: Option<bool>,
```

Comment:
Can we use an Enum to specify which version to use? We will likely to have new index format down the road.

Reply:
I initially thought of doing that first and just pass the LanceVersion to this, however, I was not sure how does it tie to the index builder. Can change it the lance version.

Reply:
+1. I would prefer a `index_file_version` parameter over a `use_new_index_format` flag.

Reply:
there are `V1` and `V3` if you are going to create that enum
V2 has been removed, no index is in V2 format now.